WeightProducer/python/MCSampleValues.py

```diff
@@ -63,493 +63,1039 @@ class MCSampleValuesHelper():
-            "CrossSection" : XSValues(XS_13TeV=7055.0),
+            "CrossSection" : XSValues(
+                XS_2016=8635, XSSource_2016="GenXSecAnalyzer",
+                XS_2017=7055, XSSource_2017="XSDB",
+                XS_2018=7025, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "QCD_Pt-300to470_MuEnrichedPt5" : {
-            "CrossSection" : XSValues(XS_13TeV=797.35),
+            "CrossSection" : XSValues(
+                XS_2016=797.3, XSSource_2016="GenXSecAnalyzer",
+                XS_2017=619.8, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=620.6, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "QCD_Pt-470to600_MuEnrichedPt5" : {
-            "CrossSection" : XSValues(XS_13TeV=59.24),
+            "CrossSection" : XSValues(
+                XS_2016=79.25, XSSource_2016="GenXSecAnalyzer",
+                XS_2017=59.24, XSSource_2017="XSDB",
+                XS_2018=59.06, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "QCD_Pt-600to800_MuEnrichedPt5" : {
-            "CrossSection" : XSValues(XS_13TeV=25.25), #2016 TuneCUETP8M1 XSDB value
+            "CrossSection" : XSValues(
+                XS_2016=25.25, XSSource_2016="XSDB",
+                XS_2017=18.18, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=18.21, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "QCD_Pt-800to1000_MuEnrichedPt5" : {
-            "CrossSection" : XSValues(XS_13TeV=4.723), #2016 TuneCUETP8M1 XSDB value
+            "CrossSection" : XSValues(
+                XS_2016=4.723, XSSource_2016="XSDB",
+                XS_2017=3.277, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=3.276, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "QCD_Pt-1000toInf_MuEnrichedPt5" : {
-            "CrossSection" : XSValues(XS_13TeV=1.613), #2016 TuneCUETP8M1 XSDB value
-        },
-        # QCD pT-hat binned: cross sections from AN2017_013_v17
-        "QCD_Pt-15to7000" : {
-            "CrossSection" : XSValues(XS_13TeV=1372000000.0), #2018 TuneCP5 GenXsecAnalyzer value
+            "CrossSection" : XSValues(
+                XS_2016=1.613, XSSource_2016="XSDB",
+                XS_2017=1.079, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=1.079, XSSource_2018="GenXSecAnalyzer",
+            ),
+        },
+        "QCD_Pt-15to7000_Flat" : {
+            "CrossSection" : XSValues(
+                XS_2016=1976000000.0, XSSource_2016="XSDB",
+                XS_2017=1370000000.0, XSSource_2017="XSDB",
+                XS_2018=1372000000.0, XSSource_2018="GenXSecAnalyzer",
+            ),
+        },
+        "QCD_Pt-15to7000_FlatP6" : {
+            "CrossSection" : XSValues(
+                XS_2016=1973000000.0, XSSource_2016="XSDB",
+            ),
+        },
+        "QCD_Pt-15to7000_Flat2017" : {
+            "CrossSection" : XSValues(
+                XS_2017=1361000000.0, XSSource_2017="XSDB",
```

Comment:
why is this different from QCD_Pt-15to7000_Flat

Reply:
Two different ways of scaling the cross section to flatten the pthat distribution.